app/(playground)/p/[agentId]/canary/actions.ts

```diff
@@ -80,6 +87,14 @@ const artifactSchema = v.object({
 			"Explanation of the Artifact and what the intention was in creating this Artifact. Add any suggestions for making it even better.",
 		),
 	),
+	usage: v.pipe(
+		v.object({
+			promptTokens: v.number(),
+			completionTokens: v.number(),
+			totalTokens: v.number(),
+		}),
+		v.description("Return from 'generateObject()' from @vercel/ai"),
+	),
```

Comment:
This will cause the LLM to generate the USAGE, so you will not get the USAGE you want to get; I think you should use the return USAGE of the streamObject, so I created a reference implementation below.

https://github.com/giselles-ai/giselle/pull/229

Reply:
thanks, that looked good!

Let us merge that into here when build error is resolved.